File: codechef/LONG/FEB19B/HMAPPY.cpp
// Suraj Pawar- ssp4all
#include <bits/stdc++.h>
# define lld long long int
int main() {
	int t;
	lld n, a, b, k, i;
	scanf("%d",&t);
	while(t--){
		lld pa = 0;
		lld pb = 0;
		scanf("%lld %lld %lld %lld",&n,&a,&b,&k);
		for(i=0; i<(n+1); i=i+a){
			if(i%a == 0 && i%b == 0){
			}
			else if(i%a == 0 && i%b != 0){
				pa++;
			}
			if((pa)>=k){
				printf("Win\n");
				break;
				// exit(0);
			}
		}
		if(pa<k){
			for(i=0; i<(n+1); i=i+b){
				if(i%a == 0 && i%b == 0){
				}
				else if(i%a != 0 && i%b == 0){
					pb++;
				}
				if((pa+pb)>=k){
					printf("Win\n");
					break;
				}
			}
		}
		if((pa+pb)<k)
			printf("Lose\n");
	}
}

File: codechef/LONG/FEB19B/HMAPPY.py
# Suraj Pawar
from fractions  import gcd
def LCM(a, b):
    return (int)(a*b)/gcd(a,b)
    
for _ in range(int(input())):
	n, a, b, k = map(int, input().split())
	pa = (int)(n/a)
	pb = (int)(n/b)
	lcm = LCM(a ,b)
	lcm = (int)(n/lcm)
# 	print(lcm)
	pa -= lcm 
	pb -= lcm
	if (pa+pb)>= k:
		print('Win')
	else:
		print('Lose')